src/Nox.Lib/Application/Providers/DefaultSystemProvider.cs

```diff
@@ -1,9 +1,24 @@
-﻿using Nox.Abstractions;
+﻿using Microsoft.AspNetCore.Http;
+using Nox.Abstractions;
 using Nox.Types;
 
 namespace Nox.Application.Providers;
 
 public class DefaultSystemProvider : ISystemProvider
 {
-    public string GetSystem() => "N/A";
+    private readonly IHttpContextAccessor _httpContextAccessor;
+
+    public DefaultSystemProvider(IHttpContextAccessor httpContextAccessor) =>
+        _httpContextAccessor = httpContextAccessor;
+
+    /// <summary>
+    /// Get the system name from the X-System-Name header
+    /// </summary>
+    /// <returns></returns>
+    public string GetSystem()
+    {
+        var result = _httpContextAccessor.HttpContext?.Request.Headers["X-System-Name"].ToString();
```

Comment:
I would suggest setting "X-System-Name" in a constant

Reply:
Desired changes were applied